install.sh

```diff
@@ -61,80 +62,81 @@ function install_macos(){
         echo "Homebrew is already installed"
     fi
     # Install the required packages
-    echo "Installing required Packages" 
+    echo "Installing required Packages"
     if [! command --version python3 &> /dev/null ]; then
 	    echo "Installing python3"
 	    brew install python@3.10
     else
 	    echo "python3 already installed."
     fi
     brew install tcl-tk python-tk
-} 
+}
 
 # Function to install for arch (and other forks like manjaro)
-function install_arch(){ 
+function install_arch() {
     echo "Installing required packages"
-    sudo pacman -S --needed python3 tk git && python3 -m ensurepip unzip || install_fail
-} 
+    sudo pacman -S --needed python3 python-pipx tk git unzip
+    python -m ensurepip || install_fail
+}
 
 # Function to install for debian (and ubuntu)
-function install_deb(){ 
+function install_deb() {
     echo "Installing required packages"
-    sudo apt install python3 python3-dev python3-tk python3-pip unzip || install_fail
-} 
+    sudo apt install python3 pipx python3-dev python3-tk python3-pip unzip || install_fail
+}
 
 # Function to install for fedora (and other forks)
-function install_fedora(){ 
-    echo "Installing required packages" 
-    sudo dnf install python3 python3-tkinter python3-pip python3-devel unzip || install_fail
-} 
+function install_fedora() {
+    echo "Installing required packages"
+    sudo dnf install python3 pipx python3-tkinter python3-pip python3-devel unzip || install_fail
+}
 
 # Function to install for centos (and other forks based on it)
-function install_centos(){
+function install_centos() {
     echo "Installing required packages"
-    sudo yum install -y python3 || install_fail
-    sudo yum install -y python3-tkinter epel-release python3-pip unzip|| install_fail
-} 
+    sudo yum install -y python3-tkinter pipx epel-release python3-pip unzip python3 || install_fail
+}
 
-function get_the_bot(){ 
-    echo "Downloading the bot" 
+function get_the_bot() {
+    echo "Downloading the bot"
     rm -rf RedditVideoMakerBot-master
     curl -sL https://github.com/elebumm/RedditVideoMakerBot/archive/refs/heads/master.zip -o master.zip
     unzip master.zip
     rm -rf master.zip
-} 
+}
 
 #install python dependencies
-function install_python_dep(){ 
+function install_python_dep() {
     # tell the user that the script is going to install the python dependencies
-    echo "Installing python dependencies" 
-    # cd into the directory
-    cd RedditVideoMakerBot-master
+    echo "Installing python dependencies"
     # install the dependencies
-    pip3 install -r requirements.txt 
-    # cd out
-    cd ..
-} 
+    DEP_LIST=./RedditVideoMakerBot-master/requirements.txt
+    if [[ -f $DEP_LIST ]]; then
+        while read -r DEPS; do
+            pipx install $DEPS
```

Comment:
Not tested on other distros, but on arch at least, pipx install packagename prints out an error for most packages:
```
No apps associated with package packagename. Try again with '--include-deps' to include apps of dependent
packages, which are listed above. If you are attempting to install a library, pipx should not be used.
Consider using pip or a similar tool instead.
```